### PR TITLE
Add babel types docs update workflow

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

Added a babel-types docs update job to the release workflow. Whenever a Babel 7 release is cut and if there are changes in the `@babel/types` docs, a website PR will be open from the website repo for the Babel 7 release.

Closes https://github.com/babel/website/pull/2819

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15953"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
If this works (we'll need to merge to be sure with the bot permissions), can we then run it as part of the release job?

**Comment by JLHwung** [comment]
> If this works (we'll need to merge to be sure with the bot permissions), can we then run it as part of the release job?

If we can add the docs generater to the regular CI, that works for me, otherwise it might block the release if there is something wrong in the script itself.

**Comment by nicolo-ribaudo** [comment]
> otherwise it might block the release if there is something wrong in the script itself. 

We can run it in parallel to the release job (just in the same workflow), similarly to how we do for the GitHub release draft creation, so that if something goes wrong it does not block the release.

**Comment by JLHwung** [comment]
Good idea. I switched my Babel fork branch and ran the workflow:

https://github.com/JLHwung/babel/actions/runs/6148201533/job/16681399277

The workflow ended with insufficient permission for push to `babel/website`, which looks good to me.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -135,6 +135,19 @@ jobs:
           # is true. Use the empry string instead.
           BABEL_8_BREAKING: ${{ needs.check-release-type.outputs.is-babel-8 == 'true' || '' }}
 
+      - name: Generate babel-types docs
+        continue-on-error: true
+        run: |
+          node ./packages/babel-types/scripts/generators/docs.js > ./build/types.md
+
+      - name: Upload babel-types docs
+        continue-on-error: true
```
Comment (nicolo-ribaudo) [inline .github/workflows/release.yml:144]: We should run this only if the previous step didn't fail, if possible

**Review by nicolo-ribaudo — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (JLHwung) [reply .github/workflows/release.yml:144]: The `continue-on-error: true` marks that if this step fails, the release job will continue regardless. If the previous step fails, the release job should stop so this step won't be executed.

**Review by liuxingbaoyu — APPROVED** [review]

**Comment by JLHwung** [comment]
@nicolo-ribaudo It seems that there is a bot permission issue (finally): https://github.com/babel/babel/actions/runs/7167820431/job/19514989666#step:4:13

**Comment by nicolo-ribaudo** [comment]
Good, I'll generate a new token :)

**Pull request merged**